cosmwasm/register-its-token.js

```diff
@@ -0,0 +1,128 @@
+require('dotenv').config();
+const env = process.env.ENV;
+const mnemonic = process.env.MNEMONIC;
+
+const { ethers } = require('hardhat');
+const { Contract, getDefaultProvider, constants: { AddressZero } } = ethers;
+const info = require(`../axelar-chains-config/info/${env}.json`);
+const tokenInfo = require(`../axelar-chains-config/info/tokens-${env}.json`);
+const IInterchainTokenService = require('@axelar-network/interchain-token-service/artifacts/contracts/interfaces/IInterchainTokenService.sol/IInterchainTokenService.json');
+const ITokenManager = require('@axelar-network/interchain-token-service/artifacts/contracts/interfaces/ITokenManager.sol/ITokenManager.json');
+const IInterchainToken = require('@axelar-network/interchain-token-service/artifacts/contracts/interfaces/IInterchainToken.sol/IInterchainToken.json')
+const fs = require('fs');
+const toml = require('toml');
+const { printInfo, printError, loadConfig, saveConfig } = require('../common');
+const { initContractConfig, prepareWallet, prepareClient, instantiateContract } = require('./utils');
+const { GasPrice } = require('@cosmjs/stargate');
+
+class TokenIterator {
+    constructor() {
+        this.tokenIndex = -1;
+        this.tokenIds = Object.keys(tokenInfo);
+        this.incrementTokenIndex();
+        this.chainIndex = -1;
+    }
+
+    tokenId() {
+        return this.tokenIds[this.tokenIndex];
+    }
+
+    token() {
+        return tokenInfo[this.tokenId()];
+    }
+
+    incrementTokenIndex() {
+        if(this.tokenIndex >= this.tokenIds.length - 1) return false;
+        this.tokenIndex++;
+        this.chainNames = Object.keys(this.token()).slice(0, -1);
+        this.chainIndex = 0;
+        return true;
+    }
+
+    get() {
+        return this.token()[this.chainName()];
+    }
+
+    chainName() {
+        return this.chainNames[this.chainIndex];
+    }
+
+    rpc() {
+        const chainName = this.chainName();
+        return env === 'mainnet' || env === 'testnet' ? RPCs.axelar_bridge_evm.find((chain) => chain.name.toLowerCase() === chainName).rpc_addr : info.chains[chainName].rpc;
```

Comment:
use a consistent rpc json structure. we have multiple environments as well, and this would allow us to reuse an rpc override mechanism in other evm scripts

Reply:
they are just provided as a command argument now, or they can be added to the `.env`